File: src/main/java/com/learnspark/plan/entity/Project.java
package com.learnspark.plan.entity;

import com.learnspark.common.entity.BaseEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.PrePersist;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.SQLRestriction;

import java.time.LocalDateTime;
import java.util.UUID;

/**
 * 学习项目实体，对应 projects 表。
 *
 * <p>按用户隔离（user_id），支持软删除（deleted_at）。
 * 使用 {@code @SQLRestriction} 自动过滤已删除记录。
 */
@Entity
@Table(name = "projects")
@SQLRestriction("deleted_at IS NULL")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Project extends BaseEntity {

    @Id
    @Column(name = "id", updatable = false, columnDefinition = "VARCHAR(36)")
    private String id;

    @Column(name = "user_id", nullable = false, columnDefinition = "VARCHAR(36)")
    private String userId;

    @Column(name = "name", nullable = false, length = 255)
    private String name;

    @Column(name = "description", columnDefinition = "TEXT")
    private String description;

    @Column(name = "goal", columnDefinition = "TEXT")
    private String goal;

    @Column(name = "daily_hours", columnDefinition = "SMALLINT")
    private Integer dailyHours;

    @Column(name = "is_ai_generated", columnDefinition = "TINYINT(1)")
    private Boolean isAiGenerated;

    @Column(name = "status", length = 20)
    private String status;

    @Column(name = "cover_color", length = 20)
    private String coverColor;

    @Column(name = "deleted_at")
    private LocalDateTime deletedAt;

    @PrePersist
    void fillDefaults() {
        if (this.id == null) {
            this.id = UUID.randomUUID().toString();
        }
        if (this.dailyHours == null) {
            this.dailyHours = 2;
        }
        if (this.isAiGenerated == null) {
            this.isAiGenerated = false;
        }
        if (this.status == null) {
            this.status = "active";
        }
        if (this.coverColor == null) {
            this.coverColor = "#18a058";
        }
    }
}

File: learnspark-kmp/server/src/main/kotlin/com/learnspark/server/domain/entity/Project.kt
package com.learnspark.server.domain.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table
import java.time.Instant
import java.time.LocalDate

/**
 * 项目实体（按文档 §11.2.3：projects 表）。
 *
 * 阶段 3.1：旧版 Vue3 数据迁移目标。
 */
@Entity
@Table(name = "projects")
class Project(
    @Id
    var id: String = "",

    @Column(name = "user_id", nullable = false)
    var userId: String = "",

    @Column(nullable = false)
    var name: String = "",

    @Column(columnDefinition = "TEXT")
    var description: String? = null,

    @Column(columnDefinition = "TEXT")
    var goal: String? = null,

    @Column(name = "cover_color", length = 20)
    var coverColor: String? = null,

    @Column(name = "daily_hours", nullable = false)
    var dailyHours: Int = 2,

    @Column(name = "is_ai_generated", nullable = false)
    var isAiGenerated: Boolean = false,

    @Column(nullable = false, length = 20)
    var status: String = "active",

    @Column(nullable = false)
    var version: Long = 0,

    @Column(name = "created_at", insertable = false, updatable = false)
    var createdAt: Instant? = null,

    @Column(name = "updated_at", insertable = false, updatable = false)
    var updatedAt: Instant? = null,
)
